File: frontend/console/src/api/vdi.js
import Axios from "@/http";

export function createVdi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vdi', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveVdi(vdiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/vdi/${ vdiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateVdi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/vdi', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteVdi(vdiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/vdi/${ vdiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteVdiWithFlag(vdiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/vdi/${ vdiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllVdis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/vdi/all').then(res => {
            resolve(res)
        })
    })
}

export function queryVdi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vdi/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/vdi.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_VDI = "/config/vdi"
const URL_ROUTER_VDI_BY_ID = "/config/vdi/:vdi-id"
const URL_ROUTER_VDI_QUERY = "/config/vdi/query"
const URL_ROUTER_VDI_ALL = "/config/vdi/all"

func VdiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_VDI, config.CreateVdi)
	r.GET(URL_ROUTER_VDI_BY_ID, config.RetrieveVdi)
	r.PUT(URL_ROUTER_VDI, config.UpdateVdi)
	r.DELETE(URL_ROUTER_VDI_BY_ID, config.DeleteVdi)
	r.PUT(URL_ROUTER_VDI_BY_ID, config.DeleteVdiWithFlag)
	return r
}
